plugins/modules/zos_job_submit.py

```diff
@@ -686,35 +699,20 @@ def run_module():
             jobId = submit_uss_jcl(src, module)
         else:
             # For local file, it has been copied to the temp directory in action plugin.
-            encoding = parsed_args.get("encoding")
-            if encoding == "EBCDIC" or encoding == "IBM-037" or encoding == "IBM-1047":
-                jobId = submit_uss_jcl(temp_file, module)
-            # 'UTF-8' 'ASCII' encoding will be converted.
-            elif (
-                encoding == "UTF-8"
-                or encoding == "ISO-8859-1"
-                or encoding == "ASCII"
-                or encoding is None
-            ):
-                (conv_rc, stdout, stderr) = module.run_command(
-                    "iconv -f ISO8859-1 -t IBM-1047 %s > %s"
-                    % (quote(temp_file), quote(temp_file_2.name)),
-                    use_unsafe_shell=True,
-                )
-                if conv_rc == 0:
-                    jobId = submit_uss_jcl(temp_file_2.name, module)
-                else:
-                    module.fail_json(
-                        msg="The Local file encoding conversion failed. Please check the source file."
-                        + stderr,
-                        **result
-                    )
+            from_encoding = encoding.get('from')
+            to_encoding = encoding.get('to')
+            (conv_rc, stdout, stderr) = module.run_command(
```

Comment:
does there need to be a check here to ensure `from_encoding` and `to_encoding` are different before running `iconv`? Not sure if it would cause any issues or not

Reply:
I am pretty sure no error happens, but not sure if `iconv` does a NOOP or actually consumes MIPS, so better to have check and do nothing if they are the same. 

```
# iconv -f IBM-1047 -t IBM-1047 hi.txt
Hello World!
````

Reply:
@blakeinate @ddimatos Yes, if both charsets are the same, no error happens. Not sure if additional MIPS are consumed though.

Reply:
I tried finding more information on if the GNU iconv implementation would check for matching character sets, was not able to find much. Tried testing on USS using `time iconv` with a large file, but in both cases (matching charsets and different charsets) the response was about the same. Wasn't much help :)

Reply:
To be fair, iconv is quite fast even for extremely large files. Hard to gauge it that way